### PR TITLE
[Merged by Bors] - sync2: fptree: don't get next in FingerprintInterval if not needed

### PR DESCRIPTION
**Pull request by ivan4th**
## Motivation

Retrieving the next item (one immediately following the fingerprinted interval) when fingerprinting an interval is only useful during slow splitting, when we need to split a range but "easy split" (involving only in-memory FPTree) fails. In other cases, the next item is not really needed and getting it may incur an unwanted database access.

`TestFPTreeManyItems` fails intermittently when it happens to fingerprint an empty range during random testing due to a problem with handling next items.

## Description

This removes unneeded retrieval of the next item and fixes `TestFPTreeManyItems`.

### CODE REVIEW

File: sync2/fptree/export_test.go [hunk 1]
```diff
@@ -7,6 +7,16 @@ import (
 
 var ErrEasySplitFailed = errEasySplitFailed
 
+func (ft *FPTree) FingerprintIntervalInternal(
```
Comment (fasmat) [inline sync2/fptree/export_test.go]: NIT:

```suggestion
func (ft *FPTree) FingerprintInternal(
```

**Review by fasmat — APPROVED** [review]

File: sync2/fptree/export_test.go (hunk 1 above)
Comment (ivan4th) [reply sync2/fptree/export_test.go]: There's already `FingerprintInterval` method which doesn't accept `needNext` argument, as `needNext` is only used internally by `FPTree`, so I named this test-only method `FingerprintIntervalInternal`.

Comment (fasmat) [reply sync2/fptree/export_test.go]: How about naming it `TestFingerprintInterval`? 🙂

Comment (ivan4th) [reply sync2/fptree/export_test.go]: It's exported b/c of blackbox testing, so that `fptree_test` package can access it

Comment (ivan4th) [reply sync2/fptree/export_test.go]: Similar to e.g. https://go.dev/src/runtime/export_test.go https://go.dev/src/net/http/export_test.go

Comment (fasmat) [reply sync2/fptree/export_test.go]: Just noticed and updated my comment. I think we can still find a better name for the method 🙂

Comment (ivan4th) [reply sync2/fptree/export_test.go]: Fixed (along with typos)

**Comment by ivan4th** [comment]
bors merge

**Comment by ivan4th** [comment]
Unrelated `TestPartition_50_50` failure:
```
    logger.go:146: 2024-12-19T21:25:05.431Z	ERROR	TestPartition_50_50	tests/partition_test.go:175	client state differs from ref state	***"client": "boot-1", "ref_client": "boot-0", "layer": 23, "client_hash": "af1349b9f5f9a1a6a0404dea36dcc9499bcb25c9adc112b7cc9a93cae41f3262", "ref_hash": "0000000000000000000000000000000000000000000000000000000000000000"***
    logger.go:146: 2024-12-19T21:25:05.431Z	ERROR	TestPartition_50_50	tests/partition_test.go:175	client state differs from ref state	***"client": "smesher-0", "ref_client": "boot-0", "layer": 23, "client_hash": "af1349b9f5f9a1a6a0404dea36dcc9499bcb25c9adc112b7cc9a93cae41f3262", "ref_hash": "0000000000000000000000000000000000000000000000000000000000000000"***
    logger.go:146: 2024-12-19T21:25:05.431Z	ERROR	TestPartition_50_50	tests/partition_test.go:175	client state differs from ref state	***"client": "smesher-1", "ref_client": "boot-0", "layer": 23, "client_hash": "af1349b9f5f9a1a6a0404dea36dcc9499bcb25c9adc112b7cc9a93cae41f3262", "ref_hash": "0000000000000000000000000000000000000000000000000000000000000000"***
```

**Comment by ivan4th** [comment]
bors merge

**Pull request closed without merging**